lib/widgets/message_list.dart

```diff
@@ -346,70 +341,62 @@ class MessageItem extends StatelessWidget {
-        right: restBorder,
-        bottom: item.isLastInBlock ? restBorder : BorderSide.none,
-      ));
-
     return StickyHeaderItem(
       allowOverflow: !item.isLastInBlock,
       header: RecipientHeader(message: message),
-      child: Column(children: [
-        DecoratedBox(
-          decoration: borderDecoration,
-          child: MessageWithPossibleSender(item: item)),
-        if (trailing != null && item.isLastInBlock) trailing!,
-      ]));
-
-    // Web handles the left-side recipient marker in a funky way:
-    //   box-shadow: inset 3px 0px 0px -1px #c2726a, -1px 0px 0px 0px #c2726a;
-    // (where the color is the stream color.)  That is, it's a pair of
-    // box shadows.  One of them is inset.
-    //
-    // At attempt at a literal translation might look like this:
-    //
-    // DecoratedBox(
-    //   decoration: ShapeDecoration(shadows: [
-    //     BoxShadow(offset: Offset(3, 0), spreadRadius: -1, color: highlightBorderColor),
-    //     BoxShadow(offset: Offset(-1, 0), color: highlightBorderColor),
-    //   ], shape: Border.fromBorderSide(BorderSide.none)),
-    //   child: MessageWithSender(message: message)),
-    //
-    // But CSS `box-shadow` seems to not apply under the item itself, while
-    // Flutter's BoxShadow does.
+      child: Stack(
+        children: [
+          MessageWithPossibleSender(item: item),
+          _UnreadMarker(isRead: message.flags.contains(MessageFlag.read)),
+        ]));
+
+  }
+}
+
+/// Widget responsible for showing the read status of a message.
+// Historically, web handled the left-side unread marker in a funky way:
+//   box-shadow: inset 3px 0px 0px -1px #c2726a, -1px 0px 0px 0px #c2726a;
+// (where the color is the stream color.)  That is, it's a pair of
+// box shadows and one of them is inset. This technique allows painting
+// a solid line inside the container without impacting the layout.
+//
+// Mobile currently uses a single inset box-shadow:
+//   box-shadow: inset 4px 0 hsl(222, 99%, 69%);
+// The animation in mobile has a 1s delay, and a 0.3s
+// transition using the ease-out curve.
+// See `src/webview/static/base.css`.
+//
+// Modern web uses a block element will a linear-gradient
+// background fill:
+//   linear-gradient(90deg, hsl(217deg 64% 59%) 30%, hsl(217deg 64% 59% / 0%))
+// Transition is 0.3s for appearing and 2s for disappearing,
+// both with an ease-out curve.
+class _UnreadMarker extends StatelessWidget {
+  const _UnreadMarker({required this.isRead});
+
+  final bool isRead;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned(
+      top: 0,
+      left: 0,
+      bottom: 0,
+      child: AnimatedOpacity(
+        opacity: isRead ? 0 : 1,
+        duration: Duration(milliseconds: isRead ? 2000 : 300),
+        curve: Curves.easeOut,
+        child: Container(
+          width: 4,
+          decoration: BoxDecoration(
+            color: const HSLColor.fromAHSL(1, 227, 0.78, 0.59).toColor()))));
```

Comment:
It came from the figma mockup, but that didn't seem like something that I could link to and no other place in the code refers to it. I've added a note pointing to the CZO discussion on this topic for this.

Reply:
Cool, sounds good. It seems like Figma does have some deep-linking capability, but a chat message is probably better to link to anyway because it gives the context of the discussion.